### DISCUSSION TITLE
10261 Rollercoaster

### DISCUSSION THREAD

**Discussion by icefloe01** [pybricks/support#578] [General]
TL;DR: Is there a way to find and import an existing Lego program that was written through the Boost app?

The rest of the story:
I built the rollercoaster and saw in the instructions about adding a hub to automate it. The only hub in stock on the Lego site was the Move Hub. They were also out of the medium motor so I got the large one. Turns out the Boost app can't control it through the Move Hub since it has serial comms on ID1 & 2 instead of a resistor. After some digging I found Philo's site (this is where I found the pin out and wiring info) and emailed him some questions. He pointed me here saying pybricks will be able to work with the motor through the Move Hub. Now I have to get a Bluetooth adaptor for my laptop since Chrome web Bluetooth won't work on mobile even after enabling the experimental features. It may still be a pointless endeavor though as I don't know if/how I'll be able to do the sound files that go with the roller coaster. Especially since I don't fully understand coding yet either. 🤣🤣

Thank you for any help,
Eric

**Comment by laurensvalk** [comment]
While you can't convert graphical programs to python directly, the rollercoaster is very simple. 

In fact, we have one 😄: https://youtu.be/XyT5AaslSaY

This one uses 2 city hubs and we adapted the design slightly, but it seems like a good idea to also make a demo for the official move hub version as well. There wouldn't be sounds, though.

Usually when designing code, it can help to describe exactly what the hub/motors/sensors should do, in words. So if you want to share what you had in mind, we can probably help you convert that to code.

**Reply by icefloe01** [reply]
I actually don't know how it works with the Lego build because I can't get the correct combination of parts that will work with the Boost app. I need either a Medium motor or a Technic hub to make a working kit. I have the Move hub, color & distance sensor, and large motor. They are out of stock on the medium motor and Technic hub. Without these parts I don't really know how it's supposed to perform.

**Reply by laurensvalk** [reply]
All motors and hubs work same, so you should be fine :)

To understand how it could work, a good start would be to just replicate how you operate the coaster manually: turning the crank.

And in case of the motor, you can attach it at the indicated point, and just let the motor run.

In code, that could look like this:

```python
from pybricks.pupdevices import Motor
from pybricks.parameters import Port, Direction
from pybricks.tools import wait

# Initialize the motor on Port C. (It doesn't matter which one you pick.)
belt_motor = Motor(Port.C, Direction.COUNTERCLOCKWISE)

# Start running the motor at 300 degrees per second
belt_motor.run(300)

# We don't need to do anything else, so just wait forever.
while True:
    wait(1000)
```

It's a good idea to try this out before you build the motor into the roller coaster. That makes it a bit easier to see what's going on.

**Reply by icefloe01** [reply]
This works and made to motor run! 😁

**Reply by icefloe01** [reply]
So I  finally got parts and then got creative. Next step is convert over to PyBricks code to load natively. I relocated the Move so I could use its internal motor to drive the chain with an axle and CV joint setup. Then I put the medium motor under the station to drive the "catch and release" system. AAA batteries don't provide enough power to drive everything so I used the 10VDC Lego supply and wired it directly into the hub. It works awesomely now!
![20220205_194230](https://user-images.githubusercontent.com/3160766/152664384-adcea2ef-b476-43f7-9a99-34b2cc82f55a.jpg)
![20220205_194256](https://user-images.githubusercontent.com/3160766/152664386-7502e72a-4e3f-4623-9108-25239ba78e06.jpg)

**Reply by laurensvalk** [reply]
Looking forward to the video!

**Reply by icefloe01** [reply]
> 

Please pardon the Lego mess and the fact that I'm not very good at making videos...🤣🤣

https://youtu.be/89SCBZgewnk
